src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_k_diffusion.py

```diff
@@ -609,6 +609,9 @@ def model_fn(x, t):
             noise_sampler = BrownianTreeNoiseSampler(latents, min_sigma, max_sigma, noise_sampler_seed)
             sampler_kwargs["noise_sampler"] = noise_sampler
 
+        if "generator" in inspect.signature(self.sampler).parameters:
```

Comment:
Allowing to pass a `generator` to the sampler so that it is relatively easy to make 1-to-1 comparison for k-diffusion samplers that require a noise, i.e. I can still use our k-diffusion pipeline but with a modified copy of the sampler that uses same noise constructor as diffusers , e.g. this script https://github.com/huggingface/diffusers/pull/5111#issuecomment-1730237336

happy to change it back if it is too hacky